src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/StatementSyntaxExtensions.cs

```diff
@@ -29,10 +29,25 @@ public static StatementSyntax GetNextStatement(this StatementSyntax statement)
             if (statement != null)
             {
                 var nextToken = statement.GetLastToken().GetNextToken();
-                return nextToken.GetAncestors<StatementSyntax>().FirstOrDefault(s => s.Parent == statement.Parent);
+                return nextToken.GetAncestors<StatementSyntax>().FirstOrDefault(s => s.Parent == statement.Parent || AreInSiblingTopLevelStatements(s, statement));
             }
 
             return null;
+
+            static bool AreInSiblingTopLevelStatements(StatementSyntax one, StatementSyntax other)
+            {
+                if (!one.IsParentKind(SyntaxKind.GlobalStatement))
+                {
+                    return false;
+                }
+
+                if (!one.IsParentKind(SyntaxKind.GlobalStatement))
+                {
+                    return false;
+                }
+
+                return one.Parent.Parent == other.Parent.Parent;
```

Comment:
```suggestion
                return one.IsParentKind(SyntaxKind.GlobalStatement) &&
                       one.IsParentKind(SyntaxKind.GlobalStatement) &&
                       one.Parent.Parent == other.Parent.Parent;
```

Reply:
💭 The last condition is redundant. Global statements are always siblings if they are in the same tree.